src/main/java/com/igot/cb/org/service/impl/OrgServiceImpl.java

```diff
@@ -184,4 +192,189 @@ public boolean isSpvRequest(String userId, List<String> requiredRoles) {
         }
         return false;
     }
+
+    public String processFrameworkCreate(String masterFramework, String orgId) {
+        String fwName = "";
+        try {
+            log.info("processFrameworkCreate started");
+            Map<String, Object> createReq = createFrameworkRequest(orgId, masterFramework);
+            Map<String, Object> request = new HashMap<>();
+            request.put(Constants.REQUEST, createReq);
+            Map<String, String> headers = new HashMap<>();
+            headers.put(Constants.X_CHANNEL_ID, orgId);
+            StringBuilder strUrl = new StringBuilder(cbServerProperties.getKnowledgeMS());
+            strUrl.append(cbServerProperties.getFrameworkCopy()).append("/").append(masterFramework);
+            log.info("Printing URL for copy: {}", strUrl);
+            log.info("Printing request: {}", request);
+            Map<String, Object> frameworkResponse = (Map<String, Object>) outboundRequestHandlerServiceImpl.fetchResultUsingPost(
+                    strUrl.toString(),
+                    request, headers);
+            if (MapUtils.isNotEmpty(frameworkResponse) && Constants.OK.equalsIgnoreCase(
+                    (String) frameworkResponse.get(Constants.RESPONSE_CODE))) {
+                Map<String, Object> result = (Map<String, Object>) frameworkResponse.get(
+                        Constants.RESULT);
+               fwName = (String) result.getOrDefault(Constants.NODE_ID, "");
+               log.info("copy framework node id: {}", fwName);
+            } else {
+                log.error("Failed to copy the framework: {}",
+                        frameworkResponse.get(Constants.RESPONSE_CODE));
+            }
+
+        } catch (Exception e) {
+            log.error("Unexpected error occurred in processFrameworkCreate", e);
+        }
+        return fwName;
+    }
+
+    public static Map<String, Object> createFrameworkRequest(String channelId, String frameworkName) {
+        Map<String, Object> framework = createFramework(channelId, frameworkName);
+        Map<String, Object> request = new HashMap<>();
+        request.put("framework", framework);
+        return request;
+    }
+
+    private static Map<String, Object> createFramework(String channelId, String frameworkName) {
+        Map<String, Object> framework = new HashMap<>();
+        StringBuilder name = new StringBuilder(channelId).append("_").append(frameworkName);
+        framework.put(Constants.NAME, name);
+        framework.put(Constants.DESCRIPTION, "Master Framework Copy");
```

Comment:
Do provide proper value for description...
something like -> Framework for Channel - channelId. Created from Master Framework.